src/htm/algorithms/Connections.hpp

```diff
@@ -36,9 +35,10 @@
 namespace htm {
 
 //TODO instead of typedefs, use templates for proper type-checking?
+//TODO profile to use better (smaller?) types
 using CellIdx   = htm::ElemSparse; // CellIdx must match with ElemSparse, defined in Sdr.hpp
-using SegmentIdx= UInt16; /** Index of segment in cell. */
-using SynapseIdx= UInt16; /** Index of synapse in segment. */ //TODO profile to use better (smaller?) types
+using SegmentIdx= unsigned char; /** Index of segment in cell. */
+using SynapseIdx= unsigned char; /** Index of synapse in segment. */
```

Comment:
the new type, only 1B!. If should overflow we have check to inform

Reply:
thanks for calling me out on this! Reverted back to UInt16, as it's more usable for many potential synapses. Also the UInt16 actually runs faster.